applications/crossbar/src/modules/cb_port_requests.erl

```diff
@@ -715,6 +724,67 @@ summary(Context) ->
 %% @doc
 %% @end
 %%--------------------------------------------------------------------
+-spec by_number(cb_context:context(), ne_binary()) -> cb_context:context().
+by_number(Context, Number) ->
+    E164 = wnm_util:to_e164(Number),
+    AccountId = cb_context:account_id(Context),
+    ViewOptions = [{'key', [AccountId, E164]}, 'include_docs'],
+    crossbar_doc:load_view(
+        ?ALL_PORT_REQ_NUMBERS
+        ,ViewOptions
+        ,cb_context:set_account_db(Context, ?KZ_PORT_REQUESTS_DB)
+        ,fun normalize_view_results/2
+    ).
+
+%%--------------------------------------------------------------------
+%% @private
+%% @doc
+%% @end
+%%--------------------------------------------------------------------
+-spec descendants_by_number(cb_context:context(), ne_binary()) -> cb_context:context().
+descendants_by_number(Context, Number) ->
+    ViewOptions = [{'keys', build_keys(Context, Number)}, 'include_docs'],
+    crossbar_doc:load_view(
+        ?ALL_PORT_REQ_NUMBERS
+        ,ViewOptions
+        ,cb_context:set_account_db(Context, ?KZ_PORT_REQUESTS_DB)
+        ,fun normalize_view_results/2
+    ).
+
+-spec build_keys(cb_context:context(), ne_binary()) -> [[ne_binary(), ne_binary()],...] | [].
```

Comment:
And you should probably compile this stuff first! Compiler complained immediately.